conversion.go

```diff
@@ -95,6 +95,56 @@ func firstDiffByteIdx(stem1 []byte, stem2 []byte) int {
 }
 
 func (n *InternalNode) InsertMigratedLeaves(leaves []LeafNode, resolver NodeResolverFn) error {
+	sort.Slice(leaves, func(i, j int) bool {
+		return bytes.Compare(leaves[i].stem, leaves[j].stem) < 0
+	})
```

Comment:
We first sort the _to be inserted leaves_ by stem, so we can group them by stem-first-byte by walking them serially.